app/src/main/java/com/github/se/icebreakrr/data/AppDataStore.kt

```diff
@@ -0,0 +1,94 @@
+package com.github.se.icebreakrr.data
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+// This file was written with the help of CursorAI
+
+/**
+ * Manages persistent app preferences using DataStore. This class handles storage and retrieval of
+ * user preferences such as:
+ * - Authentication token
+ * - Discoverability status
+ *
+ * All operations are suspending functions or return Flows to ensure safe background execution.
+ */
+open class AppDataStore(private val dataStore: DataStore<Preferences>) {
+  /**
+   * Secondary constructor that creates a DataStore instance from a Context.
+   *
+   * @param context The application context used to create the DataStore
+   */
+  constructor(context: Context) : this(context.dataStore)
+
+  companion object {
+    /**
+     * DataStore instance created using the preferencesDataStore delegate. Ensures only one instance
+     * of DataStore is created for the application.
+     */
+    private val Context.dataStore: DataStore<Preferences> by
+        preferencesDataStore(name = "app_preferences")
+
+    // Preference Keys
+    /** Key for storing the authentication token Type: String */
+    private val AUTH_TOKEN_KEY = stringPreferencesKey("auth_token")
+
+    /** Key for storing the user's discoverability preference Type: Boolean */
+    private val DISCOVERABLE_KEY = booleanPreferencesKey("is_discoverable")
```

Comment:
Nitpick: AUTH_TOKEN_KEY  & DISCOVERABLE_KEY  instances are objects created at runtime, so they do not qualify as a const val. Therefore using camelCase (authTokenKey or discoverableKey) instead of uppercase (AUTH_TOKEN_KEY) aligns more with our convention.